feat: allow defining min and max ranges for mocked numbers and strings

SampsaKaskela opened the pull request:
## Status

**READY**

## Description

Fix #2148

Adds stringMin, stringMax, numberMin and numberMax options that can be used to define default ranges for generated values if range is not defined in specification.

## Todos

- [ ] Tests
- [x] Documentation
- [ ] Changelog Entry (unreleased)

## Steps to Test or Reproduce

Build repo and use any of the above options in mock options. Values should be reflected in generated mocks.

soartec-lab reviewed (dismissed):
Thanks for your all improvements.

soartec-lab commented:
@SampsaKaskela Could you fix a conflict?

SampsaKaskela commented:
Noticed that bigInts didn't get these defaults if set since they were implemented in separate if statement. I moved the logic inside the switch case so it uses same implementation as the regular int and just switches out to bigInt if conditions are met.

I also added integer type to tests since there actually seemed to have been minor bug that if you specified int64 as format then regardless of type it would get converted to number. Not that it is common mistake anyways.
